core/src/main/java/org/apache/calcite/plan/RelOptUtil.java

```diff
@@ -159,6 +160,40 @@ public RelDataType apply(RelDataTypeField field) {
   //~ Methods ----------------------------------------------------------------
 
   /**
+   * Whether this node is a limit without sort specification.
+   */
+  public static boolean pureLimitRelNode(RelNode rel) {
+    return limitRelNode(rel) && !orderRelNode(rel);
```

Comment:
I would rename this to `isPureLimit`, and the following 3 methods similarly; the `is` makes it clear that the method is read-only.